### PR TITLE
[CSM] Add local node meta reference.

### PR DESCRIPTION
**Pull request by red-001**
A rewrite of #5505 which gets rid of the code duplication.

### CODE REVIEW

File: src/script/lua_api/l_client.cpp [hunk 1]
```diff
@@ -182,6 +185,15 @@ int ModApiClient::l_get_wielded_item(lua_State *L)
 	return 1;
 }
 
+// get_meta(pos)
+int ModApiClient::l_get_meta(lua_State *L)
+{
+	v3s16 p = read_v3s16(L, 1);
+	NodeMetadata* meta = getClient(L)->getEnv().getMap().getNodeMetadata(p);
```
Comment (SmallJoker) [inline src/script/lua_api/l_client.cpp]: `NoteMetadata *meta`

File: src/script/lua_api/l_nodemeta.cpp [hunk 2]
```diff
@@ -38,6 +38,9 @@ NodeMetaRef* NodeMetaRef::checkobject(lua_State *L, int narg)
 
 Metadata* NodeMetaRef::getmeta(bool auto_create)
 {
+	if(m_is_local)
```
Comment (SmallJoker) [inline src/script/lua_api/l_nodemeta.cpp]: `if (<condition>)`

File: doc/client_lua_api.md [hunk 3]
```diff
@@ -794,7 +796,19 @@ It can be created via `Settings(filename)`.
     * write changes to file
 * `to_table()`: returns `{[key1]=value1,...}`
 
+### NodeMetaLocalRef
```
Comment (SmallJoker) [inline doc/client_lua_api.md]: This is still `NodeMetaRef`, just as a client version.

File: doc/client_lua_api.md [hunk 4]
```diff
@@ -794,7 +796,19 @@ It can be created via `Settings(filename)`.
...
+Node metadata: reference extra data and functionality stored in a node.
+Can be obtained via `minetest.get_meta(pos)`.
 Definition tables
+
+#### Methods
+* `get_string(name)`
+* `get_int(name)`
+* `get_float(name)`
```
Comment (SmallJoker) [inline doc/client_lua_api.md]: What's the return value of each when there's no meta?

Comment (red-001) [inline doc/client_lua_api.md]: nil

**Pull request merged**